ping/ping_output.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2024-2025 Georg Pfuetzenreuter <mail+ip@georg-pfuetzenreuter.net>
+ */
+
+#include "ping.h"
+#include "stdarg.h"
+
+void ping_print_int(struct ping_rts *rts, char *msg, char *json_key, int json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_INT, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+}
+
+void ping_print_uint(struct ping_rts *rts, char *msg, char *json_key, unsigned int json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_UINT, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+}
+
+void ping_print_str(struct ping_rts *rts, char *msg, char *json_key, char *json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_STR, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+}
+
+inline void ping_print_version(struct ping_rts *rts) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_STR, "version", PACKAGE_VERSION);
+		print_json_packet(rts);
+	} else {
+		printf(PACKAGE_VERSION);
```

Comment:
You replaced `printf(IPUTILS_VERSION("ping"));` with `printf(PACKAGE_VERSION)` why? That caused to print just git hash instead of `ping from iputils 20240905-42-ge6cc75f\n`. Please put it back.

Before:
```C
$ ./builddir/ping/ping -V
ping from iputils 20240905-42-ge6cc75f
libcap: yes, IDN: yes, NLS: yes, error.h: yes, getrandom(): yes, __fpending(): yes
```
With your patch:
```C
$ ./builddir/ping/ping -V
20240905-44-gbe6d3dblibcap: yes, IDN: yes, NLS: yes, error.h: yes, getrandom(): yes, __fpending(): yes
```

Reply:
Thanks for info, I finally understand what you mean and replied there.

Reply:
Thanks!